src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java

```diff
@@ -298,11 +304,17 @@ private Map<MediaEntity, Path> downloadAndSaveMediaToWorkspace(List<MediaEntity>
     @Override
     public Path saveMediaToWorkspace(MediaEntity mediaEntity) throws IOException {
         UUID id = getMediaLocation(mediaEntity).orElseThrow(
-            () -> new RuntimeException(String.format("Could not locate UUID for media: %s", mediaEntity.getId()
-            )));
+                () -> new RuntimeException(String.format("Could not locate UUID for media: %s", mediaEntity.getId()
+                )));
 
-        log.debug("Downloading audio blob for {} from unstructured datastore", id);
-        BinaryData binaryData = getUnstructuredAudioBlob(id);
+        BinaryData binaryData = null;
+        try {
+            log.debug("Downloading audio blob for {} from unstructured datastore", id);
+            binaryData = getUnstructuredAudioBlob(id);
+        } catch (BlobStorageException blobStorageException) {
+            log.debug("Downloading audio blob for {} from dets datastore", id);
+            binaryData = getDetsAudioBlob(id);
+        }
```

Comment:
I wonder whether it would be better to be more specific here, i.e. downloading from DETS only if we get an exception that is the equivalent in Azure of a 404. 
This would prevent a scenario where the file is in unstructured, but download fails for some reason, and we do an unnecessary download from DETS which will fail because the file is not there

Reply:
@richardbull - can we get a second opinion on this please

Reply:
We don't need these checks here, its already been done in Facade.

Reply:
@hemantasharma1129 facade checks for unstructured -> dets -> arm. are we ok to use it here (since we are not checking arm)?

Reply:
I don't particularly like the `id` variable name that was there before, we chop and change between using `location` and `externalLocation` quite a bit. I prefer externalLocation, only because that corresponds to our table `external_object_directory.external_location`. In Azure Storage it's just a uuid `Name` property.

I'm not sure about the `BlobStorageException` currently as I haven't done much work in that area. I see what @mario-paniccia is saying though. As a tiny compromise, we could possibly add the `blobStorageException` to the debug line?